server/channels/app/import_functions.go

```diff
@@ -1671,6 +1746,15 @@ func (a *App) importMultiplePostLines(rctx request.CTX, lines []imports.LineImpo
 		return 0, model.NewAppError("importMultiplePostLines", "app.post.overwrite.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
 	}
 
+	// Update thread memberships for posts that were overwritten. Here some of the memberships
+	// can be brand new, needs to be updated or an older membership should not get updated.
+	// MaintainMembership method has some logic within to handle those decisions. Unfortunately
+	// some application code leaked to the store layer here, which should be revisited when there
+	// is resource (eg. time, human or maybe AI).
+	if _, sErr := a.Srv().Store().Thread().MaintainMultipleFromImport(threadMembersToOverwriteList); sErr != nil {
```

Comment:
Important thing to note here is that the `importMultiplePostsThreshold` is set to 1000. So while there is limit on posts, there is no limit on the thread memberships of a post. So ideally, we should be batch input this method as well. But if you can tested this on a large enough data without issues, then I guess it should be fine. cc @streamer45